client/scripts/stores/RecentListingStore.ts

```diff
@@ -0,0 +1,206 @@
+    const { topicName, stageName } = params;
+    if (topicName) {
+      return this._fetchState.topicInitialized[topicName];
+    } else if (stageName) {
+      return this._fetchState.stageInitialized[stageName];
+    } else {
+      return this._fetchState.allThreadsInitialized;
+    }
+  }
+
+  public initializeListing(params: IListingParams) {
+    const { topicName, stageName } = params;
+    if (topicName) {
+      this._fetchState.topicInitialized[topicName] = true;
+    } else if (stageName) {
+      this._fetchState.stageInitialized[stageName] = true;
+    } else {
+      this._fetchState.allThreadsInitialized = true;
+    }
+  }
+
+  // Subpage depletion status, i.e. "Have all threads of this type been fetched?"
+
+  public isDepleted(params: IListingParams): boolean {
+    const { topicName, stageName } = params;
+    if (topicName) {
+      return this._fetchState.topicDepleted[topicName];
+    } else if (stageName) {
+      return this._fetchState.stageDepleted[stageName];
+    } else {
+      return this._fetchState.allThreadsDepleted;
+    }
+  }
+
+  public depleteListing(params: IListingParams) {
+    const { topicName, stageName } = params;
+    if (topicName) {
+      this._fetchState.topicDepleted[topicName] = true;
+    } else if (stageName) {
+      this._fetchState.stageDepleted[stageName] = true;
+    } else {
+      this._fetchState.allThreadsDepleted = true;
+    }
+  }
+
+  // Cutoff date, i.e. "Up to what archival point have threads been fetched?"
+
+  public getCutoffDate(params: IListingParams): moment.Moment {
+    const { topicName, stageName } = params;
+
+    if (topicName) {
+      return this._fetchState.topicCutoffDate[topicName];
+    } else if (stageName) {
+      return this._fetchState.stageCutoffDate[stageName];
+    } else {
+      return this._fetchState.allThreadsCutoffDate;
+    }
+  }
+
+  public setCutoffDate(params: IListingParams, cutoffDate: moment.Moment) {
+    const { topicName, stageName } = params;
+    if (topicName) {
+      this._fetchState.topicCutoffDate[topicName] = cutoffDate;
+    } else if (stageName) {
+      this._fetchState.stageCutoffDate[stageName] = cutoffDate;
+    } else {
+      this._fetchState.allThreadsCutoffDate = cutoffDate;
+    }
+  }
+
+  // Filter function to determine inclusion of threads on listing page
+
+  private _isBeforeCutoff(
+    params: IListingParams,
+    thread: OffchainThread
+  ): boolean {
+    const { topicName, stageName, pinned } = params;
+    const listingCutoff = topicName
+      ? this._fetchState.topicCutoffDate[topicName]
+      : stageName
+      ? this._fetchState.stageCutoffDate[stageName]
+      : this._fetchState.allThreadsCutoffDate;
+    const lastUpdate = thread.lastCommentedOn || thread.createdAt;
+    return (pinned && thread.pinned) || +lastUpdate >= +listingCutoff;
+  }
+
+  // When topics are deleted, the threads associated with them must be updated
+
+  public removeTopic(topicName: string) {
```

Comment:
Should this also remove from `fetchState`?

Reply:
Yes, good catch!